src/sentry_value.c

```diff
@@ -1058,14 +1090,54 @@ sentry_event_value_add_stacktrace(sentry_value_t event, void **ips, size_t len)
     sentry_value_t stacktrace = sentry_value_new_object();
     sentry_value_set_by_key(stacktrace, "frames", frames);
 
-    sentry_value_t thread = sentry_value_new_object();
-    sentry_value_set_by_key(thread, "stacktrace", stacktrace);
+    return stacktrace;
+}
+
+static sentry_value_t
+sentry__get_values_list(sentry_value_t parent, const char *key)
```

Comment:
This is an upsert since it creates the value. Consider calling it `__get_or_insert_values_list` or `__ensure_values_list`.